aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -893,11 +966,25 @@ def route(
         cors: Optional[bool] = None,
         compress: bool = False,
         cache_control: Optional[str] = None,
+        middleware: Optional[List[Callable]] = None,
     ):
         def register_route(func: Callable):
             # Convert methods to tuple. It needs to be hashable as its part of the self._routes dict key
             methods = (method,) if isinstance(method, str) else tuple(method)
-            self._routes[(rule, methods, cors, compress, cache_control)] = func
+
+            tuple_key = (rule, methods, cors, compress, cache_control)
+
+            # Collate Middleware for routes
+            if middleware is not None:
+                for handler in middleware:
+                    self._routes_with_middleware[tuple_key] = self._routes_with_middleware.get(tuple_key, []).append(
```

Comment:
can we break this into two expressions to reduce cognitive load? (append in a separate one)

Reply:
already done in a mypy patch - its not type safe like this.

refactored to:
```
            if middleware is not None:
                for handler in middleware:
                    if self._routes_with_middleware.get(tuple_key) is None:
                        self._routes_with_middleware[tuple_key] = [handler]
                    else:
                        self._routes_with_middleware[tuple_key].append(handler)
```